ably-ios/ARTRest.m

```diff
@@ -200,10 +207,20 @@ - (void)time:(void(^)(NSDate *time, NSError *error))callback {
     return nil;
 }
 
-- (void)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback {
-    NSParameterAssert(query.limit < 1000);
-    NSParameterAssert([query.start compare:query.end] != NSOrderedDescending);
-    
+- (BOOL)stats:(ARTStatsQuery *)query callback:(void (^)(ARTPaginatedResult *, NSError *))callback error:(NSError **)errorPtr {
```

Comment:
I can't comment on whether this is a suitable style for Obj-C, however typically in our APIs we raise exceptions directly when calling the Rest library as it is mostly a synchronous styled API, whereas with Realtime everything is more `async` and thus we pass an error to the callback unless the method being called is deterministically invalid i.e. you called the method with invalid arguments.